src/distraction_manager.cpp

```diff
@@ -0,0 +1,216 @@
+                        configurable_distractions[currentLine].second.c_str() );
+
+        // Draw horizontal line and corner pieces of the table
+        for( int x = 0; x < 78; x++ ) {
+            if( x == 60 ) {
+                mvwputch( w_header, point( x, iHeaderHeight - 2 ), c_light_gray, LINE_OXXX );
+                mvwputch( w_header, point( x, iHeaderHeight - 1 ), c_light_gray, LINE_XOXO );
+            } else {
+                mvwputch( w_header, point( x, iHeaderHeight - 2 ), c_light_gray, LINE_OXOX );
+            }
+        }
+
+        wnoutrefresh( w_header );
+
+        // Clear table
+        for( int y = 0; y < iContentHeight; y++ ) {
+            for( int x = 0; x < 79; x++ ) {
+                if( x == 60 ) {
+                    mvwputch( w, point( x, y ), c_light_gray, LINE_XOXO );
+                } else {
+                    mvwputch( w, point( x, y ), c_black, ' ' );
+                }
+            }
+        }
+
+        draw_scrollbar( w_border, currentLine, iContentHeight, number_of_distractions, point( 0,
+                        iHeaderHeight + 1 ) );
+
+        calcStartPos( startPosition, currentLine, iContentHeight, number_of_distractions );
+
+        for( int i = startPosition; i < number_of_distractions; ++i ) {
+            const nc_color line_color = i == currentLine ? hilite( c_white ) : c_white;
+            const nc_color status_color = distractions_status[i] ? c_light_green : c_red;
+            const std::string status_string = distractions_status[i] ? _( "Enabled" ) : _( "Disabled" );
+
+            // Print distraction types
+            mvwprintz( w, point( 1, i - startPosition ), line_color, "%s",
+                       configurable_distractions[i].first.c_str() );
+
+            // Print "Enabled/Disabled" text
+            mvwprintz( w, point( 62, i - startPosition ), status_color, "%s", status_string );
+        }
+
+        wnoutrefresh( w_header );
+        wnoutrefresh( w_border );
+        wnoutrefresh( w );
+    } );
+
+    while( true ) {
+        ui_manager::redraw();
+
+        const std::string currentAction = ctx.handle_input();
+
+        if( currentAction == "QUIT" ) {
+            break;
+        }
+
+        if( currentAction == "UP" ) {
+            if( currentLine > 0 ) {
+                --currentLine;
+            } else {
+                currentLine = number_of_distractions - 1;
+            }
+        } else if( currentAction == "DOWN" ) {
+            if( currentLine == number_of_distractions - 1 ) {
+                currentLine = 0;
+            } else {
+                ++currentLine;
+            }
+        } else if( currentAction == "CONFIRM" ) {
+            // This will change status color and status text
+            distractions_status[currentLine] = !distractions_status[currentLine];
+
+            // This will actually toggle enabled/disabled status of distractions
+            switch( currentLine ) {
+                case 0:
```

Comment:
It could be worth converting these numbers into enums.

Reply:
I thought about it. The thing is, there is already one enum with all distraction types, I just didn't want to create another one, almost exactly the same as original, just without few types.